lib/galaxy_test/api/test_groups.py

```diff
@@ -76,10 +76,15 @@ def test_update(self):
 
         group_id = group["id"]
         updated_name = f"group-test-updated-{self.dataset_populator.get_random_name()}"
-        update_payload = {
-            "name": updated_name,
-        }
-        update_response = self._put(f"groups/{group_id}", data=update_payload, admin=True, json=True)
+        update_response = self._put(f"groups/{group_id}", data={"name": updated_name}, admin=True, json=True)
+        self._assert_status_code_is_ok(update_response)
+
+        # Update with another user
+        another_user_id = None
+        with self._different_user():
+            another_user_id = self.dataset_populator.user_id()
+        assert another_user_id is not None
```

Comment:
Do you need to reset the group after the test ?

Reply:
Hmmm... that could explain the interaction between tests... I keep forgetting about that, thanks! :sweat_smile: 

If that is the case, does this mean that adding a user (B) to a group where another user (A) is a member, will grant user B access to the datasets of user A? Or am I reading incorrectly the tests?

Reply:
> does this mean that adding a user (B) to a group where another user (A) is a member, will grant user B access to the datasets of user A?

I'll answer myself, it is because the group also has the private role of user (A), so adding another user to the group will grant that user (B) access to the private role of user (A). Which probably makes sense and it is how permissions work.